pkg/bucket/bucket_controller.go

```diff
@@ -71,7 +84,7 @@ func (b *BucketListener) Add(ctx context.Context, inputBucket *v1alpha1.Bucket)
 
 	if bucket.Spec.BucketClassName == "" {
 		err = errors.New(fmt.Sprintf("BucketClassName not defined for bucket %s", bucket.ObjectMeta.Name))
-		klog.V(3).ErrorS(err, "BucketClassName not defined")
+		b.recordEvent(inputBucket, v1.EventTypeWarning, "MissingBucketClassName", "BucketClassName was not defined in the Bucket")
```

Comment:
One of the things Rook does is codify the event types  in it's API so that applications using Rook programmatically can see the definitions. Should we do this with COSI as well? This would be "MissingBucketClassName" here.

At any rate, it seems best to use a const definition for MissingBucketClassName.

---

As a different thought? Is this maybe just a special case of InvalidBucketClass? What if the event type were just InvalidBucketClass, and the message to the right will explain exactly what is invalid?

Reply:
I am wondering - are event types just constants, or generated via kubebuilder? If kubebuilder is the direction to go, this will probably require some additional rethinking.

Reply:
In Rook, we have just made them constants.

The field is called "reason", and following the k8s API trend, I imagine something like:

```go
const (
    MissingBucketClassNameEventReason = "MissingBucketClassName"
    
    InvalidBucketInfoEventReason = "InvalidBucketInfo"
    
    WhateverElseEventReason = "..."
)
```

Reply:
Yeah, please use typed constants instead of string literals like "MissingBucketClassName"